wandb/wandb_agent.py

```diff
@@ -1,517 +1,165 @@
-import json
-import logging
-import multiprocessing
+# -*- coding: utf-8 -*-
+"""Agent - Agent object.
+
+Manage wandb agent.
+
+"""
+
+from __future__ import print_function
+
 import os
-import platform
-import signal
 import socket
-import subprocess
-import sys
 import time
-import traceback
 
-import six
-from six.moves import queue
 import wandb
 from wandb import util
+from wandb import wandb_sdk
 from wandb.apis import InternalApi
 from wandb.lib import config_util
-import yaml
-
-logger = logging.getLogger(__name__)
-
-
-class AgentError(Exception):
-    pass
-
-
-class AgentProcess(object):
-    """Launch and manage a process."""
 
-    def __init__(
-        self, env=None, command=None, function=None, run_id=None, in_jupyter=None
-    ):
-        self._popen = None
-        self._proc = None
-        self._finished_q = multiprocessing.Queue()
-        self._proc_killed = False
-
-        if command:
-            if platform.system() == "Windows":
-                kwargs = dict(creationflags=subprocess.CREATE_NEW_PROCESS_GROUP)
-            else:
-                kwargs = dict(preexec_fn=os.setpgrp)
-            self._popen = subprocess.Popen(command, env=env, **kwargs)
```

Comment:
you seem to be killing the `wandb agent` command line variant which uses subprocess to execute scripts.   How will the python loop work for that case?

I think you want to keep the old subprocess way and split out the new python loop based approach for wandb.agent()